### DISCUSSION TITLE
Are past prompts in a single conversation shared back each time?

### DISCUSSION THREAD

**Discussion by isaacblum** [open-webui/open-webui#664] [Q&A] [answered]
First off, this product you have invested in is fantastic. Thank you to everyone who supports it!

Question: I have used the OpenAI ChatGPT and Azure's equivalent version, using GPT4 models. I often have it help me update Python code or configurations. When I work with the chatbot, I paste in the relevant section or whole .py or script. Then at the top of the initial chat prompt, I ask, "Can you help optimize the script?" In most cases, it provides recommendations and then updates either part of the script or the whole script.

Issue: when using Ollama on a separate server (with GPU) and using Ollama Web UI in a docker container, the setup works for basic things like asking it to summarize text or check spelling. What I'm finding to be strange is that often times in similar scenarios as above, it will only respond with written recommendations, but not code updates. And if I ask it to update the script that I have previously shared, it will say something like, "When you're ready please share me the script," among various other scenarios like that.

It's as if the conversation history isn't being retained and sent after my initial prompt. I feel like I'm missing something really obvious, so I'm hoping someone can help me out.

Models Tested with: "mixtral:latest (24.6GB)" | "codellama:13b (6.9GB)"

Thanks again

p.s. sorry if this is not an ollama-webui question and more of a ollama or the LLM's themselfs

**Comment by justinh-rahb** [comment, marked as answer]
I'd say you seem to be hitting the context window limit. Mixtral's should be set to 32k in the modelfile, have you perhaps modified the `Context Length` or `Max Tokens` parameters in Settings > Advanced? CodeLlama only has a context size limit of 4096 tokens, which is quite small if you're working with code.

The context window is the fixed-size sequence that language models consider when generating output. If the input sequence exceeds this limit, the model may not be able to take into account earlier parts of the sequence, leading to errors or inconsistencies in the generated output.

**Reply by isaacblum** [reply]
@justinh-rahb - I've seen you reply to many folks' questions... you rock!

I wasn't paying close attention to those sliders. I thought the same thing, so I just dragged it to max. But it only shows 16000 or 16k. Changing both of them to 32k is netting results more consistent with what I would expect.

I knew this was something minor I was missing. Thank you so much again!

**Reply by justinh-rahb** [reply]
@isaacblum Glad you've got it sorted out! I'd actually recommend just leaving them both at "Default", which will defer to the `n_ctx` parameter that is set by the model, and return as many tokens as the model feels will adequately answer your question. I only use the `Max Tokens` parameter if I know I want a short response (400 tokens equates to around 100 words for example).
